### PR TITLE
Stricter apply bindings

### PR DESCRIPTION
**Pull request by orangejenny**
screen shot 2015-12-03 at 6 19 14 pm

@dannyroberts / @proteusvacuum / @benrudolph 

If y'all like this, I'll do a sweep to use it more broadly.

### CODE REVIEW

**Comment by benrudolph** [comment]
seems reasonable to me. obviously adds the dep on jquery but that's fine with me

**Comment by orangejenny** [comment]
resolved merge conflict

**Comment by dannyroberts** [comment]
Ooo this is even nicer than what I was thinking of. It's very intuitive that this method would throw an exception if called on an empty jQuery node set.

**Comment by proteusvacuum** [comment]
Nice!
Would it be worth also issuing a warning if the nodeset this is called on has more than one matching node?

**Comment by orangejenny** [comment]
ooh, good thought

**Comment by orangejenny** [comment]
@proteusvacuum https://github.com/dimagi/commcare-hq/commit/b1730ca0eeac97cd80350789f9a9789e5c0be25e

File: corehq/apps/style/static/style/js/hq.helpers.js [hunk 1]
```diff
@@ -122,3 +122,9 @@ $.fn.enableButton = function () {
 };
 
 
+$.fn.koApplyBindings = function (context) {
+    if (!this.length) {
+        throw "No element passed to koApplyBindings";
```
Comment (millerdev) [inline corehq/apps/style/static/style/js/hq.helpers.js]: [Best practice](http://stackoverflow.com/a/27501348/10840) is to `throw new Error("...")`. Browsers like Chrome will display a stack trace, but other tools like phantomjs will not, which can make debugging more difficult.

Comment (millerdev) [inline corehq/apps/style/static/style/js/hq.helpers.js]: ignore. I see you fixed it later.

**Pull request merged**